src/did/src/codec.rs

```diff
@@ -15,3 +15,117 @@ pub fn bincode_encode<T: serde::Serialize>(item: &T) -> Vec<u8> {
 pub fn bincode_decode<'a, T: serde::Deserialize<'a>>(bytes: &'a [u8]) -> T {
     bincode::deserialize(bytes).expect("failed to deserialize item with bincode")
 }
+
+pub mod macro_utils {
+    use ic_stable_structures::BoundedStorable;
+
+    /// Returns if `T` is a of fixed size
+    pub const fn is_fixed_size<T: BoundedStorable>(_: &T) -> bool {
+        T::IS_FIXED_SIZE
+    }
+
+    /// Returns if `T`'s max size
+    pub const fn get_max_size<T: BoundedStorable>(_: &T) -> u32 {
+        T::MAX_SIZE
+    }
+}
+
+/// Encodes several BoundedStorable items into a Vec<u8>
+#[macro_export]
+macro_rules! encode_fixed_storables {
```

Comment:
I think it would be more idiomatic to have a macro that automatically derives BoundedStorable for a struct. Have tried this approach?

Reply:
That took some time... But now I've switched to procedural macro